src/wallet/wallet.cpp

```diff
@@ -1714,7 +1723,7 @@ bool CWallet::AddToWalletIfInvolvingMe(const CTransaction& tx, const CBlock* pbl
         bool fExisted = mapWallet.count(tx.GetHash()) != 0;
         if (fExisted && !fUpdate) return false;
         auto sproutNoteData = FindMySproutNotes(tx);
-        auto saplingNoteDataAndAddressesToAdd = FindMySaplingNotes(tx);
+        auto saplingNoteDataAndAddressesToAdd = FindMySaplingNotes(tx, chainActive.Height());
```

Comment:
We shouldn't be using `chainActive.Height()` unless `pblock` is null and so we can't get `pblock`'s height. This should be changed to

1. Grab the height from `pblock` if available; or else
2. assume that the transaction is in the mempool and so `LOCK(cs_main)` and obtain `chainActive.Height()` instead.

Reply:
i refactored `AddToWalletIfInvolvingMe()` to take in `pindex`, so we can get the height from `pindex->nHeight` if `pblock` is available.

Reply:
this has been changed to pass `nHeight` directly to `AddToWalletIfInvolvingMe()`. (see https://github.com/zcash/zcash/pull/4578#discussion_r449715802)